src/VisualStudio/Core/Def/Implementation/VirtualMemoryNotificationListener.cs

```diff
@@ -104,5 +116,46 @@ public int OnBroadcastMessage(uint msg, IntPtr wParam, IntPtr lParam)
 
             return VSConstants.S_OK;
         }
+
+        private bool ShouldTurnOffFullSolutionAnalysis(long availableMemory)
+        {
+            // conditions
+            // 1. if available memory is less than the threshold and 
+            // 2. if full solution analysis memory monitor is on (user can set it off using registery, when he does, we will never show info bar) and
+            // 3. if our full solution analysis option is on (not user full solution analysis option, but our internal one) and
+            // 4. if infobar is never shown to users for this solution
```

Comment:
Why does 4. matter? Are we suggesting we will never implicitly turn of full solution analysis for the following sequence of steps:

1. Low VM situation, FSD turned off with info bar shown.
2. It got turned back on either by user explicitly doing it from error list OR infobar OR opening new solution.
3. After some time, we hit low VM situation again - will this cause VS to crash with OOM?